### ISSUE TITLE
DenseNet generated image embeddings have very small variance and are sparse

### ISSUE THREAD

**Issue by zlenyk** [pytorch/vision#1904]
I'm working with pretrained DenseNet161 model to learn and generate image embeddings.
I consider output of adaptive_avg_pool2d layer image embedding. I noticed that it's very sparse and has small variance. If we consider a batch of 64 images, variance of the same feature over all images in a batch is ~2e-6. Also the feature values are just very small (~1e-3). This is for pretrained model, if I start fine-tuning the model, some features quickly become 0 across all images in the batch.
Surprisingly, this is not the case for resnet. Running the same training for resnet50 results in expected features of size ~[0, 2] and variance around 1.
This is stunning difference for me. Not sure if there's some problem with implementation or this is just DenseNet works. Please let me know if  you have similar experiences.

**Comment by fmassa** [comment]
Hi,

I have never used DenseNet myself, so I wouldn't know the answer to your question.
I would first check though that your densenet model is outputting correct class predictions -- maybe you are passing the wrong input normalization, which makes almost all the activations go to zero due to ReLU?

**Comment by filoliufei** [comment]
> 我正在使用预训练的DenseNet161模型来学习和生成图像嵌入。
> 我考虑了adaptive_avg_pool2d图层图像嵌入的输出。我注意到它非常稀疏且变化很小。如果我们考虑一批64张图像，则同一特性在一批图像中的方差约为2e-6。同样，特征值也很小（〜1e-3）。这是针对预训练模型的，如果我开始对模型进行微调，则批处理中所有图像上的某些功能会很快变为0。
> 令人惊讶的是，resnet并非如此。对resnet50进行相同的训练会导致预期特征的大小约为[0，2]，方差约为1。
> 这对我来说是惊人的区别。不确定实现是否存在问题，或者这仅仅是DenseNet的工作。如果您有类似的经验，请告诉我。

Hello, I'm learning densenet121on Imagenet, the test accuracy after importing the pre training parameters is about 0. Are you referring to the pre training weight parameters in the article?

**Comment by fmassa** [comment]
@filoliufei are you using the image normalization as expected by the model? Also, do you get expected accuracies with ResNet50?

**Comment by filoliufei** [comment]
Yes. The problem has been solved. Thank you.

**Comment by datumbox** [comment]
This issue appears to be stale, so I'm closing it. If you think it should remain open, feel free to reopen it.
